src/Utilities/Compiler/Options/SymbolNamesWithValueOption.cs

```diff
@@ -21,7 +21,9 @@ internal sealed class SymbolNamesWithValueOption<TValue> : IEquatable<SymbolName
         internal const char WildcardChar = '*';
 
         public static readonly SymbolNamesWithValueOption<TValue> Empty = new SymbolNamesWithValueOption<TValue>();
+#pragma warning disable CA1805 // Do not initialize unnecessarily - Removing this suppression causes CS0649 (Field is never assigned to, and will always have its default value)
```

Comment:
FYI @stephentoub

Reply:
I guess the other option would be this?

```
internal static readonly KeyValuePair<string?, TValue> NoWildcardMatch
    = KeyValuePair.Create<string?, TValue>(null, default);
```

Reply:
Sure, but should still fix the analyzer false positive here. I'll leave the pragma in case we can get rid of CA1805 here, and eventually get faded unnecessary pragmas after https://github.com/dotnet/roslyn/pull/44848

Reply:
I'm not entirely sure what to do here.  The CS0649 warning is a little strange because, with or without the `new SymbolNamesWithValueOption<TValue>()`, the field is being assigned its default value.  And the whole purpose of CA1805 is to flag cases where the assignment isn't needed (or, worse, if there were another static field on this type being assigned to a non-default value, the compiler would actually output this default assignment rather than eliding it).

Reply:
Sounds good - let me apply CA1805 and suppress CS0649 here.

Reply:
> Sure, but should still fix the analyzer false positive here

It's not really a false positive: CA1805 is flagging a case where the assignment is needed, and removing the assignment will improve the IL.  If anything, it's CS0649 that's the false positive here, IMO.

Reply:
It occurs to me the best fix here is actually:
```C#
internal static KeyValuePair<string, TValue> NoWildcardMatch => default;
```
I don't think there's a good reason to store this in a field at all.

Reply:
Sure, I can do so. Another interesting fact is the CS0649 is generated only on subset of projects that contain this shared project file. @sharwell would you know why that would be the case?

![image](https://user-images.githubusercontent.com/10605811/86182198-5afc1080-bae4-11ea-91a0-f3b1b6703702.png)

Reply:
The property is `internal`. Projects with one or more IVTs disable CS0649 on `internal` fields because they can be set by code outside the scope of the current compilation.

Reply:
> because they can be set by code outside the scope of the current compilation

Even when it's readonly?